### DISCUSSION TITLE
Spytify 1.12

### DISCUSSION THREAD

**Discussion by jwallet** [jwallet/spy-spotify#462] [Release]
- Added an option to force Spotify to skip already recorded tracks #154 
- Hide mp3 settings when wav format is selected #451 
- Fixed "Record everything" setting, was not able to record podcast and/or skip those already recorded #457 
- New option to prefix filenames with the album track number #460

<img width="728" alt="image" src="https://user-images.githubusercontent.com/23088305/211070517-21daacb1-3a92-42aa-94cf-0431ba178f65.png">

<hr /><em>This discussion was created from the release <a href='https://github.com/jwallet/spy-spotify/releases/tag/1.12'>Spytify 1.12</a>.</em>

**Comment by sideprojectslab** [comment]
Hello, I noticed that often Spytify cuts the head of a song, most likely due to Spotify only signalling the start of a new song after the audio has already begun. Would it make sense to have a pre-record window of, say, 5 seconds and, when spotify signals a new song also include the portion of the pre-recorded window that contains audio? For tracks that fade into each other one could also use this pre-record window to make sure the two tracks have no overlapping content, by basically keeping the last track "unfinished" until the next one has been signalled, and then using some logic to splice between the two...

**Reply by DonJema** [reply]
I would also welcome such a function - a **pre / post buffer function**, because tracks are often cut off at the beginning or end.

**Reply by jwallet** [reply]
https://github.com/jwallet/spy-spotify/issues/359

**Comment by jeremyhallden** [comment]
Hi @jwallet ,
It seems that it is not possible to select 320kbps anymore. If selected it has no effect on recording. And if I restart the app, the setting is reset to the default 160kbps again.
Or am I missing something here? :)

**Reply by jeremyhallden** [reply]
@jwallet Any idea on why this is happening? (Is broken in both 1.11 & 1.12) 
Should I open an issue instead?

**Reply by jwallet** [reply]
It's already fixed, https://github.com/jwallet/spy-spotify/issues/456, you can build the app yourself or wait until I release the next version

**Reply by jeremyhallden** [reply]
Great! Thank you :)

**Comment by rpraleigh** [comment]
Downloaded and unzipped but when I run the executable I get this error:
![image](https://user-images.githubusercontent.com/2387268/216113825-0284e664-3741-4baf-8c9f-e676bf380173.png)

**Reply by jwallet** [reply]
running in a vm? or in windows remote ? you should have at least one audio device, plug headphones in your sound card. If the default audio device is bluetooth, you might want to test a different one.

**Comment by JonnyGee** [comment]
Hi @jwallet, just started testing out Spytify 1.12 and all seems good except despite having Spotify Premium and the MP3 settings in Spytify set to 320 Kbps, the resulting file is always around 150 Kbps. I've tried running as admin but out of ideas further than that. Any ideas?

**Comment by johnric** [comment]
I'm a new user, just downloaded v1.12 and installed the VB-Audio Virtual Cable driver.

I've changed my system sound settings to use the Virtual Cable for both input and output.
Spytify settings likewise changed to specify VB-Audio Virtual Cable as Audio Device.

I have the "Record Everything" option checked in Advanced settings because I want to record a podcast.

But I always get the error:
"Spytify cannot record if it doesn't use the same audio endpoint device than Spotify. "

Your FAQ page has no specific instructions for starting a recording. I assume it is supposed to pick up an audio stream automatically when the Start button has been activated.

What am I missing?

**Reply by fableman73** [reply]
This version is broken, not saving any files... DL 1.11 that version works.

**Reply by bloxjw** [reply]
@jwallet i start the record and it says recording but when i stop recording it doesn't say recorded and i don't get it in my output folder.

**Comment by aladeeni** [comment]
fantastic app

**Comment by floppyman1980** [comment]
Thank you to the hero who made this. U r great.
I managed to download songs that was 320kbps and there is very little, i say little difference in the music quality.
Of all the places i cold find a free Spotify music and it was called Spytify.

**Comment by CelestialHappyorMoody-a11y** [comment]
Hey, today (feb 2) I tried to open the Spytify like normal and it sent me a pop up and the pop up itself is not working- how or can it get fix? Other that I love Spytify.
<img width="407" height="233" alt="image" src="https://github.com/user-attachments/assets/ed6874a3-1d40-4b50-9d80-9bc8efc648b9" />
